File: AnalisisNumerico/Java/SustiProgre.java
package AnalisisNumerico;

public class SustiProgre {
	private static double xn = 0;
	public static double[] sustiProgre(double L[][],double b[], int d) {
		double z[] = new double[d];
		for(int i=0;i<d;i++) {
			for(int j=0;j<d;j++) {
				xn =xn + z[j]*L[i][j];
			}
			z[i] = b[i]-xn;
			xn=0;
		}
		for(int k =0;k<d;k++) {
			System.out.println("z"+(k+1)+" :"+z[k]);
		}
		return z;
	}
}

File: AnalisisNumerico/Sistema de ecuaciones lineales/SustiProgre.m
function z = SustiProgre(L, b)
  #Con la matriz L nxn y el vector b nx1
  #Se ejecuta para calcular el vector z nx1
  d = length(b);
  xn = 0;
  Uaux = zeros(d);
  z = Uaux(:,1);
  for i=1:d
    for j=1:d
      xn = xn + z(j)*L(i,j);
    endfor
    z(i) = b(i)-xn;
    xn = 0;
  endfor
endfunction